backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/diagram/ViewDiagramDescriptionConverter.java

```diff
@@ -572,4 +560,19 @@ private IDiagramContext getDiagramContext(VariableManager variableManager) {
         return variableManager.get(IDiagramContext.DIAGRAM_CONTEXT, IDiagramContext.class).orElse(null);
     }
 
+    /**
+     * A structure representing the current state of the conversion.
+     *
+     * @author fbarbin
+     */
+    private class ConvertedState {
```

Comment:
I personally do not have any issue with this kind of private helper classes, but:

* if should be `static` unless it needs a hidden reference to the enclosing instance;
* it should still follow our coding practices (`final` fields, getters);
* in this case, instead of being a "dumb" wrapper for two collections, maybe it should expose a more meaningful and slightly high-level API which matches how the collections are used in the converter. See for example `org.eclipse.sirius.components.diagrams.renderer.DiagramRenderingCache` which handles a similar problem (keep track of the mutable state during a multi-step process).

If the class becomes more than a dumb wrapper, it probably deserves to be its own top-level type.

Reply:
Personnaly, I've never really seen a good reason why we NEED to be absolutely sure that nobody won't ever be able to use a class.

For me it's a mix of trying to predict the future and trying to hide stuff and I don't see why. A proper class which does one thing and well and which could be reused somewhere one day is always better in my mind.

Apart from very specific cases like creating an API which should be manipulated in a very specific way like our builder patterns used to prevent ˋnull` on specific fields or the spring security http configuration pattern (where you can write ˋhttp.crsf().disable().and()...`), I don't see the point.

Reply:
I extracted this class with a package visibility.

Reply:
And why do you need this package visibility? Why would you NEED to prevent anyone from seeing this?

I still don't understand what's your reason to try to hide some code from others (and from who?)

Reply:
> For me it's a mix of trying to predict the future and trying to hide stuff and I don't see why.
> ...
> could be reused somewhere one day

So, you're trying to predict the future too, just with a different default assumption :)

One argument for hiding things unless explicitly needed is to reduce our API surface and minimize future breakage. As soon as we make something public, it becomes API and any change to it will need to be documented and possibly break some users who started depending on it.

Reply:
Since we don't use OSGi, everything is public by default. We can't hide from that. Any class even at the very bottom of our code stack which has no reason to be manipulated by anyone else than us has to be public because we need it in another part of our code.

Should we make `EditingContextEventProcessor` a private class? If so should we move it's code inside of `EditingContextEventProcessorFactory`? Or should we switch it to a package visibility? Should we have all of our code in a single package to hide everything from the outer world?

This is a plain old Java project, everything is public by default. That's the default assumption in Java.